### ISSUE TITLE
Luppp and footcontroller experience and tips???!

### ISSUE THREAD

**Issue by ArneSchulz** [openAVproductions/openAV-Luppp#172]
Hello, everyone,
I want to use luppp for a performance. I need a footcontroller to work with luppp!
I need four loops and four scenes! Is it possible to switch ( loop/scene) between 1,2,3,4 Loops and 1,2,3,4 scenes by using the same footcontrols? (ICON g-board?)
It would be great to have two inputs and four outputs (or stereo) and a headphone for monitoring. (Roland octacapture?)
As well it would be great to have a controller for vol, mute, and pan.
everything should be as small as possible, could a tablet work for that purpose?
Any Ideas or comments?
Thanx Arne

![loppp setting](https://cloud.githubusercontent.com/assets/25999596/23295552/8312f0d4-fa71-11e6-96bc-0ee814fb954e.gif)

**Comment by geraldmwangi** [comment]
Hi, You'd need to assign different controls to the loops and scenes, 
otherwise you active multiple thing (a loop on scene 2 and complete 
scene 1) at once.

I don't think you want that?

For per track output add "enablePerTrackOutput":    1, to luppp.prfs.

At the moment luppp isn't capable of multiple inputs and I cant say when 
that will happen.

March is going to see some rework of the internal audio/time engine. 
I'll evaluate if multiple inputs is easy to integrate.

Lg Gerald

**Comment by ArneSchulz** [comment]
Thanx for the interest and reply!

Is it really impossible to have two commands on one control nob? I worked a lot with abelton and there it is possible. Maybe by a script that tells which kind of action has to get started?

>   Hi, You'd need to assign different controls to the loops and scenes, otherwise you active multiple thing (a loop on scene 2 and complete scene 1) at once.

**Comment by georgkrause** [comment]
@ArneSchulz there is a feature comming which will allow you to script your midi actions as you want. but not ready yet. 
Until than you could work with the special Clip function. you assign a special clip with right click -> special and than you can trigger the action with a midi controller. if this is an option, we can talk about!

in general, your wished controller seems pretty specific and custom, it might be a good idea to built it on you own. again, if you need any input where to start, we can talk about ;)

**Comment by harryhaaren** [comment]
Hi @ArneSchulz,

Yes its possible to write bindings right now with the Generic MIDI backend for Luppp: For example, a grid press event can be done by sending a note on (0x90, or 144 decimal), with a note number of 40, to track 0 scene 2, using the following snippet:
https://github.com/openAVproductions/openAV-Luppp/blob/master/resources/controllers/akai_apc_mini.ctlr#L18 

This is the *current* way to do things. What George mentions above is a new scripting backend that is in development, but not ready for real-world use.

In short, right now it is achievable to do what you're hoping to do, it just requires some effort in scripting / testing your setup. Hopefully that will become easier in future due to the CtlraScript feature.

Cheers, -Harry

**Comment by georgkrause** [comment]
@harryhaaren but he wants to use one switch to control different clips, if i understand correctly

**Comment by ArneSchulz** [comment]
Hallo everyone,
I think georgkrause is right! 
One nob to switch between loops and scene. so I can choose in which scene I want to work in 1, 2, 3, or 4. I hit the scene / loop button again and I am in the loop mode and can choose loop 1, 2, 3 or 4. 
I´m a musician, guitarist and do some livelooping. for me it seems this would be the most effective way to handle livelooping with luppp. 
I got the idea from this guy:
https://www.youtube.com/watch?v=z3H_KAHWdyw
info@sonnit.co.uk. 
Thax for discussing this idea, it is more then I expected!
I tried a lot of free loopsoftware mobius, superlooper, but I love luppp and think it has the most potential of all!
greetings arne

**Comment by georgkrause** [comment]
@ArneSchulz in the video there is a knob for each track and there is no scene change as far as i see. So you would also need to use the 4 track buttons. you dont need the mode button because the first hit records, the second hit stops recording and then you can play/stop a clip. you could use this button to go through the scenes. you start at scene 1, one hit and it goes to scene 2 and so on...

**Comment by sub26nico** [comment]
I use luppp with a Roland FC200 + a Boss FS-6(to change octave in mode note).
 I can rec/play/stop the loops on many scenes.
I can't control the scene(play/stop) because midi binding doesn't work well on my laptop,
but work better on my pc. On the laptop, I had to edit the config file to can use
the foot controller. it work well now.

**Comment by georgkrause** [comment]
@ArneSchulz still some questions or can i close this?

**Comment by ArneSchulz** [comment]
Hello Georg,

if you want to you can close this questions!

Best regards

Arne


Arne Schulz

Barnerstr.25

22765 Hamburg

arne-schulz@hotmail.de

Tel: 0176 700 78 471


________________________________
Von: Georg Krause <notifications@github.com>
Gesendet: Sonntag, 19. März 2017 19:24
An: openAVproductions/openAV-Luppp
Cc: ArneSchulz; Mention
Betreff: Re: [openAVproductions/openAV-Luppp] Luppp and footcontroller experience and tips???! (#172)


@ArneSchulz<https://github.com/ArneSchulz> still some questions or can i close this?
